apps/rand.c

```diff
@@ -95,8 +97,44 @@ int rand_main(int argc, char **argv)
     argc = opt_num_rest();
     argv = opt_rest();
     if (argc == 1) {
+        int factoridx = 0;
+        int scale = 0;
+        /*
+         * iterate over the value and check to see if there are
+         * any non-numerical chars
+         */
+        while (argv[0][factoridx]) {
+            if (!isdigit(argv[0][factoridx])) {
+                switch(argv[0][factoridx]) {
+                case 'k':
+                    scale = 1;
+                    break;
+                case 'm':
+                    scale = 2;
+                    break;
+                case 'g':
+                    scale = 3;
+                    break;
+                default:
+                    goto opthelp; 
+                }
+                break;
+            }
+            factoridx++;
+        }
+        if (scale != 0)
+            argv[0][factoridx] = '\0';
+
         if (!opt_long(argv[0], &num) || num <= 0)
             goto opthelp;
+
+        if (scale != 0) {
+            scaled_num = num * (1 << (scale * 10));
```

Comment:
Ah, so `scale` is binary, rather than decimal. Why not make it `10`, `20` or `30` directly? Really, it's the log2 of the scale, of course

Reply:
Sure, I thought it was more readable this way, but I can use 10,20,30 directly

Reply:
yes, I've modified it to use use 10/20/30